pages/agent/v3/signed_pipelines.md

```diff
@@ -0,0 +1,130 @@
+  <ul>
+    <li>It had to be installed on every agent instance, leading to more configuration</li>
+    <li>It only supported symmetric HS256 signatures, meaning that every verifier could also sign uploads</li>
+    <li>It couldn't sign matrix steps</li>
+  </ul>
+
+  This newer version of pipeline signing is built right into the agent, and addresses all of these limitations. Being built into the agent, it's easier to configure and use.
+
+  Many thanks to Seek.com.au, who we collaborated with on the older version of the tool, and whose prior art has been instrumental in the development of this newer version.
+</details>
+
+### What we actually sign
+
+As mentioned above, Signed Pipelines establishes that important aspects of steps haven't been changed since pipeline upload. In more detail, we sign the following aspects of steps:
+
+- The command
+- Environment variables _as defined in the pipeline YAML_ - environment variables set by the agent, by hooks, or by the user's shell are not signed and can potentially override the environment that a step's command is started with.
+
+- Any plugins used by the step, and their configurations
+- The matrix configuration
+  + The matrix configuration is signed as a whole, rather than each individual matrix job. This means that the signature will be the same for each job in the matrix. When signatures are verified for matrix jobs, we double-check that the job we've received is a valid construction of the matrix, and that the signature matches the matrix configuration.
+- The step's key, if present [WIP]
+
+Additionally, we also sign the following aspects of the build for each job:
+
+- The pipeline's UUID [WIP]
+- The organization UUID [WIP]
+- The build number [WIP]
+- The repo the build is running for [WIP]
+- The commit the build is running for [WIP]
+
+
+## Enabling signed pipelines on your agents
+
+### 1. Generating a key pair
+Behind the scenes, Signed Pipelines uses [JSON Web Signing (JWS)](https://datatracker.ietf.org/doc/html/rfc7797) to generate its signatures. This means that you'll need to generate a [JSON Web Key Set (JWKS)](https://datatracker.ietf.org/doc/html/rfc7517) to sign and verify your pipelines with, then configure your agents to use those keys.
+
+Luckily, the Buildkite Agent has you covered! There's a JWKS generation tool built into the agent, which you can use to generate a key pair for you. To run it, you'll need to [install the agent on your machine](/docs/agent/v3/installation), and then run:
+
+```bash
+buildkite-agent tool keygen --alg <algorithm> --key-id <key-id>
+```
+
+replacing `<algorithm>` and `<key-id>` with the algorithm you want to use, and the key ID you want to use. For example, to generate an EdDSA key pair with a key ID of `my-key-id`, you'd run:
```

Comment:
Is there a link we could include for EdDSA? Maybe just https://en.wikipedia.org/wiki/EdDSA ?